Source/BoundaryConditions/PML_RZ.cpp

```diff
@@ -0,0 +1,252 @@
+#endif
+    for ( amrex::MFIter mfi(*E_fp[0], amrex::TilingIfNotGPU()); mfi.isValid(); ++mfi )
+    {
+        amrex::Array4<amrex::Real> const& Et_arr = E_fp[1]->array(mfi);
+        amrex::Array4<amrex::Real> const& Ez_arr = E_fp[2]->array(mfi);
+        amrex::Array4<amrex::Real> const& Bt_arr = B_fp[1]->array(mfi);
+        amrex::Array4<amrex::Real> const& Bz_arr = B_fp[2]->array(mfi);
+
+        amrex::Array4<amrex::Real> const& pml_Et_arr = pml_E_fp[1]->array(mfi);
+        amrex::Array4<amrex::Real> const& pml_Bt_arr = pml_B_fp[1]->array(mfi);
+
+        // Get the tileboxes from Efield and Bfield so that they include the guard cells
+        // They are all the same, cell centered
+        amrex::Box tilebox = amrex::convert((*E_fp[1])[mfi].box(), E_fp[0]->ixType().toIntVect());
+
+        // Box for the whole simulation domain
+        amrex::Box const& domain = m_geom->Domain();
+        const int nr_domain = domain.bigEnd(0);
+
+        // Set tilebox to only include the upper radial cells
+        const int nr_damp = m_ncell;
+        int nr_damp_min;
+        if (m_do_pml_in_domain) {
+            nr_damp_min = nr_domain - nr_damp;
+        } else {
+            nr_damp_min = nr_domain;
+        }
+        tilebox.setSmall(0, nr_damp_min + 1);
+
+        amrex::ParallelFor( tilebox, E_fp[0]->nComp(),
+            [=] AMREX_GPU_DEVICE (int i, int j, int k, int icomp)
+            {
+                const amrex::Real rr = static_cast<amrex::Real>(i - nr_damp_min);
+                const amrex::Real wr = rr/nr_damp;
+                const amrex::Real damp_factor = std::exp( -4._rt * cdt_over_dr * wr*wr );
+
+                // Substract the theta PML fields from the regular theta fields
+                Et_arr(i,j,k,icomp) -= pml_Et_arr(i,j,k,icomp);
+                Bt_arr(i,j,k,icomp) -= pml_Bt_arr(i,j,k,icomp);
+                // Damp the theta PML fields
+                pml_Et_arr(i,j,k,icomp) *= damp_factor;
+                pml_Bt_arr(i,j,k,icomp) *= damp_factor;
+                // Add the theta PML fields back to the regular theta fields
+                Et_arr(i,j,k,icomp) += pml_Et_arr(i,j,k,icomp);
+                Bt_arr(i,j,k,icomp) += pml_Bt_arr(i,j,k,icomp);
+                // Damp the z fields
+                Bz_arr(i,j,k,icomp) *= damp_factor;
+                Ez_arr(i,j,k,icomp) *= damp_factor;
+            });
+    }
+}
+
+std::array<MultiFab*,2>
+PML_RZ::GetE_fp ()
+{
+    return {pml_E_fp[0].get(), pml_E_fp[1].get()};
+}
+
+std::array<MultiFab*,2>
+PML_RZ::GetB_fp ()
+{
+    return {pml_B_fp[0].get(), pml_B_fp[1].get()};
+}
+
+void
+PML_RZ::FillBoundary ()
+{
+    FillBoundaryE();
+    FillBoundaryB();
+}
+
+void
+PML_RZ::FillBoundaryE ()
+{
+    FillBoundaryE(PatchType::fine);
+}
+
+void
+PML_RZ::FillBoundaryE (PatchType patch_type)
+{
+    if (patch_type == PatchType::fine && pml_E_fp[0] && pml_E_fp[0]->nGrowVect().max() > 0)
+    {
+        const auto& period = m_geom->periodicity();
```

Comment:
I would be in favor of using the explicit types as much as possible, if you agree:
```suggestion
        const amrex::Periodicity& period = m_geom->periodicity();
```

Reply:
Fixed